src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -1007,7 +1007,11 @@ int wm_vuldet_fill_report_oval_data(sqlite3 *db, sqlite3_stmt *stmt, agent_softw
         sqlite3_bind_text(stmt, 1, report->cve, -1, NULL);
         sqlite3_bind_text(stmt, 2, target, -1, NULL);
 
-        ref_count = 0;
+        for (ref_count = 0; report->references && report->references[ref_count]; ++ref_count) {
```

Comment:
Considering that in this part of the code, we are adding some extra information to the `report` structure apart from the NVD information that was already added, we need to check whether that and if so, count the number of references in order to append the new ones.
If we don't count the references, we will not be appending information to the array and as consequence, we will lose the memory references of the information already added, causing in this way a memory leak.